docs/rustdoc-release.md

```diff
@@ -0,0 +1,59 @@
+# Substrate Rustdoc Release Process
+
+Substrate rustdoc sits on the [paritytech/substrate `gh-pages` branch](https://github.com/paritytech/substrate/tree/gh-pages) and is accessible at https://paritytech.github.io/substrate.
```

Comment:
Was about to ask what sources rustdocs on substrate.dev. But hey, it's great that now it [points](https://docs.substrate.io/rustdocs/) here!